staging/src/k8s.io/client-go/tools/cache/thread_safe_store_test.go

```diff
@@ -166,6 +169,124 @@ func TestThreadSafeStoreIndexingFunctionsWithMultipleValues(t *testing.T) {
 	assert.NoError(compare("baz", []string{}))
 }
 
+func TestThreadSafeStoreRV(t *testing.T) {
+	t.Run("Initial state", func(t *testing.T) {
+		store := NewThreadSafeStore(Indexers{}, Indices{}).(*threadSafeMap)
+		if rv := store.RV(); rv != "" {
+			t.Errorf("Expected initial RV to be \"\", got %q", rv)
+		}
+	})
+
+	t.Run("Add and Update", func(t *testing.T) {
+		store := NewThreadSafeStore(Indexers{}, Indices{}).(*threadSafeMap)
+
+		// Add obj with RV "10"
+		store.Add("key1", &metav1.ObjectMeta{ResourceVersion: "10"})
+		if rv := store.RV(); rv != "10" {
+			t.Errorf("Expected RV to be \"10\", got %q", rv)
+		}
+
+		// Add obj with lower RV "5"
+		store.Add("key2", &metav1.ObjectMeta{ResourceVersion: "5"})
+		if rv := store.RV(); rv != "10" {
+			t.Errorf("Expected RV to remain \"10\" after adding lower RV, got %q", rv)
+		}
+
+		// Add obj with same RV "10"
+		store.Add("key3", &metav1.ObjectMeta{ResourceVersion: "10"})
+		if rv := store.RV(); rv != "10" {
+			t.Errorf("Expected RV to remain \"10\" after adding same RV, got %q", rv)
+		}
+
+		// Add obj with higher RV "20"
+		store.Add("key4", &metav1.ObjectMeta{ResourceVersion: "20"})
+		if rv := store.RV(); rv != "20" {
+			t.Errorf("Expected RV to be \"20\", got %q", rv)
+		}
+
+		// Update obj with lower RV "15"
+		store.Update("key4", &metav1.ObjectMeta{ResourceVersion: "15"})
+		if rv := store.RV(); rv != "20" {
+			t.Errorf("Expected RV to remain \"20\" after updating with lower RV, got %q", rv)
+		}
+
+		// Add obj with invalid RV "abc" (should be ignored due to comparison error)
+		store.Add("key5", &metav1.ObjectMeta{ResourceVersion: "abc"})
+		if rv := store.RV(); rv != "20" {
+			t.Errorf("Expected RV to remain \"20\" after adding invalid RV, got %q", rv)
+		}
+
+		// Add obj with no RV
+		store.Add("key6", &metav1.ObjectMeta{ResourceVersion: ""})
+		if rv := store.RV(); rv != "20" {
+			t.Errorf("Expected RV to remain \"20\" after adding empty RV, got %q", rv)
+		}
+
+		// Add non-meta object
+		store.Add("key7", "just a string")
+		if rv := store.RV(); rv != "20" {
+			t.Errorf("Expected RV to remain \"20\" after adding non-meta object, got %q", rv)
+		}
+	})
+
+	t.Run("Replace", func(t *testing.T) {
+		store := NewThreadSafeStore(Indexers{}, Indices{}).(*threadSafeMap)
+		store.Add("key1", &metav1.ObjectMeta{ResourceVersion: "10"})
+
+		if rv := store.RV(); rv != "10" {
+			t.Fatalf("Setup failed, expected RV \"10\", got %q", rv)
+		}
+
+		items := map[string]interface{}{
+			"key2": &metav1.ObjectMeta{ResourceVersion: "30"},
+			"key3": &metav1.ObjectMeta{ResourceVersion: "40"},
```

Comment:
Don't want to assume implementation so would be good to expand the items to make it clear that a max is picked.
Can you change order so make sure you just not take the last? Adding more elements would also work.